python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -702,5 +737,29 @@ def fillCoadd(self, dcrModels, skyInfo, tempExpRefList, weightList, calibration=
             self.assembleMetadata(coaddExposure, tempExpRefList, weightList)
             coaddUtils.setCoaddEdgeBits(model[skyInfo.bbox].mask, model[skyInfo.bbox].variance)
             coaddExposure.setMaskedImage(model[skyInfo.bbox])
+            if mask is not None:
+                coaddExposure.setMask(mask)
             dcrCoadds.append(coaddExposure)
         return dcrCoadds
+
+    def calculateGain(self, modelIter):
+        """Calculate the gain to use for the current iteration.
+
+        Parameters
+        ----------
+        modelIter : `int`
+            The current iteration of forward modeling.
+
+        Returns
+        -------
+        gain : `float`
+            Relative weight to give the new solution when updating the model.
+        """
+        if self.config.useProgressiveGain:
+            baseGain = 0.5
+            if modelIter < 2:
+                return baseGain
```

Comment:
baseGain is only used here, you could just return 0.5 or only define it inside this block if you want it named, but naming it does not seem to explain why the value is 0.5 any more so maybe err to less lines of code